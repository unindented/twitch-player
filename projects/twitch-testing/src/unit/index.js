import { ThemeProvider } from "@twitch-player/themes/dist/context";
import { light as lightTheme } from "@twitch-player/themes/dist/themes";
import React from "react";
import { render } from "react-testing-library";

const customRender = (node, ...options) => {
  return render(
    <ThemeProvider theme={lightTheme}>{node}</ThemeProvider>,
    ...options
  );
};

const fireResizeEvent = (el, { left = 0, top = 0, width = 0, height = 0 }) => {
  Object.defineProperties(window.HTMLElement.prototype, {
    offsetLeft: { get: () => left },
    offsetTop: { get: () => top },
    offsetWidth: { get: () => width },
    offsetHeight: { get: () => height },
  });

  const evt = document.createEvent("Event");
  evt.initEvent("resize", true, true);
  el.dispatchEvent(evt);
};

export * from "react-testing-library";

export { customRender as render, fireResizeEvent };
