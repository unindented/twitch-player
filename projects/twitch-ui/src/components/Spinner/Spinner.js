import PropTypes from "prop-types";
import React, { memo } from "react";
import { ActivityIndicator, StyleSheet, View } from "react-native";
import { useTheme } from "../../hooks";

const Spinner = memo(({ testID = "spinner" }) => {
  const theme = useTheme();

  return (
    <View style={styles.root} testID={testID}>
      <ActivityIndicator
        color={theme.colors.spinner}
        size={theme.layout.spinnerSize}
      />
    </View>
  );
});

Spinner.propTypes = {
  testID: PropTypes.string,
};

const styles = StyleSheet.create({
  root: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
});

export default Spinner;
