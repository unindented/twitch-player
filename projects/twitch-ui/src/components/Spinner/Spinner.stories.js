import { storiesOf } from "@storybook/react";
import React from "react";
import Spinner from ".";

storiesOf("Spinner", module).add("default", () => <Spinner />);
