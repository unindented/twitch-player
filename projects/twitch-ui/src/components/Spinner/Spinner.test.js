import { render } from "@twitch-player/testing/dist/unit";
import React from "react";
import Spinner from "./Spinner";

describe("Spinner", () => {
  let instance;

  beforeEach(() => {
    instance = render(<Spinner />);
  });

  it("renders a spinner", () => {
    expect(instance.getByTestId("spinner")).toMatchSnapshot();
  });
});
