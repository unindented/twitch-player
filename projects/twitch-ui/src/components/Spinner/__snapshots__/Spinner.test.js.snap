// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders a spinner 1`] = `
<div
  class="rn-alignItems-1awozwy rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexGrow-16y2uox rn-flexShrink-1wbh5a2 rn-flexBasis-1ro0kt6 rn-flexDirection-eqz5dr rn-justifyContent-1777fci rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
  data-testid="spinner"
>
  <div
    aria-valuemax="1"
    aria-valuemin="0"
    class="rn-alignItems-1awozwy rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-justifyContent-1777fci rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
    role="progressbar"
  >
    <div
      class="rn-alignItems-1oszu61 rn-animationDuration-17bb2tj rn-animationIterationCount-1muvv40 rn-animationName-1twrwyb rn-animationTimingFunction-1ldzwu0 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
      style="height: 36px; width: 36px;"
    >
      <svg
        height="100%"
        viewBox="0 0 32 32"
        width="100%"
      >
        <circle
          cx="16"
          cy="16"
          fill="none"
          r="14"
          stroke-width="4"
          style="stroke: #6441a4; opacity: 0.2;"
        />
        <circle
          cx="16"
          cy="16"
          fill="none"
          r="14"
          stroke-width="4"
          style="stroke: #6441a4; stroke-dasharray: 80; stroke-dashoffset: 60;"
        />
      </svg>
    </div>
  </div>
</div>
`;
